### PR TITLE
Maintainance Updates

### PR DESCRIPTION
**Pull request by kevinkirkup**
* Updated to golang 1.14
* Updated acceptance tests to use 0.12 syntax
* Corrected documentation: name_regex has change to filter
  The documentation was updated but the example HCL was incorrect.

### CODE REVIEW

**Review by joshpoth — APPROVED** [review]
lgtm 🚋🚋🚋

File: pureport/resource_pureport_google_cloud_connection_test.go [hunk 1]
```diff
@@ -94,8 +94,8 @@ resource "pureport_google_cloud_connection" "main" {
   name = "%s"
   speed = "50"
 
-  location_href = "${data.pureport_locations.main.locations.0.href}"
-  network_href = "${data.pureport_networks.main.networks.0.href}"
+  location_href = data.pureport_locations.main.locations.0.href
+  network_href = data.pureport_networks.main.networks.0.href
 
   primary_pairing_key = "${google_compute_interconnect_attachment.main.0.pairing_key}"
```
Comment (Try75) [inline pureport/resource_pureport_google_cloud_connection_test.go:100]: Do you still need the "${...}" formatting here?

Comment (kevinkirkup) [reply pureport/resource_pureport_google_cloud_connection_test.go:100]: no, missed that one.

Comment (kevinkirkup) [reply pureport/resource_pureport_google_cloud_connection_test.go:100]: Updated in #25

**Pull request merged**